core/templates/pages/preferences-page/preferences-page.component.spec.ts

```diff
@@ -31,63 +26,202 @@ import { TranslatorProviderForTests } from 'tests/test.extras';
 require('pages/preferences-page/preferences-page.component.ts');
 
 describe('Preferences Controller', function() {
-  var ctrl, $httpBackend;
+  var ctrl = null;
+  var $httpBackend = null;
+  var $rootScope = null;
+  var $q = null;
+  var $timeout = null;
+  var $uibModal = null;
+  var CsrfService = null;
+  var UserService = null;
+  var userInfo = {
+    getUsername: () => 'myUsername',
+    email: () => 'myusername@email.com'
+  };
+  var mockWindow = {
+    location: {
+      reload: () => {}
+    }
+  };
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('$window', mockWindow);
+  }));
 
-  beforeEach(function() {
-    angular.mock.module('oppia');
-  });
   beforeEach(angular.mock.module('oppia', function($provide) {
     var ugs = new UpgradedServices();
     for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
       $provide.value(key, value);
     }
   }));
 
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value(
-      'AudioLanguageObjectFactory', new AudioLanguageObjectFactory());
-    $provide.value(
-      'AutogeneratedAudioLanguageObjectFactory',
-      new AutogeneratedAudioLanguageObjectFactory());
-    $provide.value('UserInfoObjectFactory', new UserInfoObjectFactory());
-  }));
-  beforeEach(
-    angular.mock.module('oppia', TranslatorProviderForTests));
+  beforeEach(angular.mock.module('oppia', TranslatorProviderForTests));
 
   beforeEach(angular.mock.inject(function($injector, $componentController) {
-    var $http = $injector.get('$http');
-
     $httpBackend = $injector.get('$httpBackend');
+    $q = $injector.get('$q');
+    $rootScope = $injector.get('$rootScope');
+    $timeout = $injector.get('$timeout');
+    $uibModal = $injector.get('$uibModal');
+
+    CsrfService = $injector.get('CsrfTokenService');
+    UserService = $injector.get('UserService');
+
+    spyOn(CsrfService, 'getTokenAsync').and.returnValue(
+      $q.resolve('sample-csrf-token'));
+
+    ctrl = $componentController('preferencesPage', {
+      $rootScope: $rootScope
+    });
+
+    spyOn(UserService, 'getUserInfoAsync').and.returnValue(
+      $q.resolve(userInfo));
     $httpBackend.expectGET('/preferenceshandler/data').respond({
       can_receive_email_updates: false,
       can_receive_editor_role_email: true,
       can_receive_feedback_message_email: true
     });
 
-    ctrl = $componentController('preferencesPage', null, {
-      $http: $http,
-      AlertsService: {}
-    });
-    // Refer: https://www.codelord.net/2017/01/09/
-    // unit-testing-angular-components-with-%24componentcontroller/
-    // Angular and $componentController does not take care of
-    // $onInit lifecycle hook, so we need to call it explicitly.
     ctrl.$onInit();
+    $rootScope.$apply();
+    $httpBackend.flush();
   }));
 
-  it('should show that editor role notifications checkbox is true by default',
```

Comment:
Why are all these tests deleted?

Reply:
I've refactored this file so I've just changed the order of `it` blocks. You can see this block was not deleted at all.